### PR TITLE
feature: Show crosshair when hovering on track

### LINKED ISSUE #312: Improve track plot navigation
## Use Case

- [ ] Show current time point on Plotly via a larger dot
- [x] Allow clicking on certain time points to jump to that frame, like how the scatter plot handles this.

## Acceptance Criteria

_Please describe how you know this is done_

## Details

_Please provide any helpful specifications_

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
Closes #312, "improve track plot navigation".

This implements part 2 of the original spec by adding in a crosshair that shows the currently-hovered value.

*Estimated review size: small, 10-15 minutes*

Solution
========
- Changed `setTime` in `Plotting` to `updateLayout`, which renders both the dotted line (representing the current time) and the crosshair.
- Added state to track the currently hovered object in the plot (in `PlotWrapper`.)
- Added a loading spinner while the current frame is being loaded in `PlotTab`.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-465/
2. Open any dataset and select a track. Hover over the line.

Screenshots (optional):
-----------------------
image

### CODE REVIEW

**Comment by rugeli** [comment]
The preview looks good to me! Just a small thought: should we hide the leading zero if there is one, for a cleaner value presentation? In the case shown in the screenshot, maybe display `756.4724` instead of `0,756.4724`?  
Screenshot 2024-11-08 at 2 43 23 PM

**Review by rugeli — APPROVED** [review]
LGTM!

**Comment by ShrimpCryptid** [comment]
> The preview looks good to me! Just a small thought: should we hide the leading zero if there is one, for a cleaner value presentation? In the case shown in the screenshot, maybe display `756.4724` instead of `0,756.4724`? Screenshot 2024-11-08 at 2 43 23 PM

Oh, I think in that case it's showing the `X, Y` coordinates? So the X-axis value is 0 and the Y-axis value is 756.472.

**Comment by rugeli** [comment]
> Oh, in that case it's showing the `X, Y` coordinates. So the X-axis value is 0 and the Y-axis value is 756.472.

lol my bad! Thanks for the correction!

File: src/components/PlotWrapper.tsx [hunk 1]
```diff
@@ -3,6 +3,8 @@ import React, { ReactElement, useEffect, useMemo, useRef, useState } from "react
 
 import { Dataset, Plotting, Track } from "../colorizer";
 
+import { TrackPlotLayoutConfig } from "../colorizer/Plotting";
```
Comment (frasercl) [inline src/components/PlotWrapper.tsx:6]:
```suggestion
import type { TrackPlotLayoutConfig } from "../colorizer/Plotting";
```

File: src/components/PlotWrapper.tsx [hunk 2]
```diff
@@ -37,10 +40,24 @@ export default function PlotWrapper(inputProps: PlotWrapperProps): ReactElement
...
+      const featureData = props.dataset?.getFeatureData(props.featureKey);
+      const hoveredObjectTime = props.dataset?.getTime(hoveredObjectId);
+      if (featureData && hoveredObjectTime) {
+        const featureValue = featureData.data[hoveredObjectId];
+        config.hover = {
+          x: hoveredObjectTime,
+          y: featureValue,
+        };
+      }
+    }
```
Comment (frasercl) [inline src/components/PlotWrapper.tsx:58]: Entirely optional thought: consider reordering this `if` and the definition of `config`, adding a `let hover: { x: number, y: number } | undefined = undefined;` (type extra optional), and using `hover` in the definition of `config`:

```typescript
const config: TrackPlotLayoutConfig = {
  time: props.frame,
  hover,
};
```

Makes the structure of the value you're creating a bit more obvious, without referencing the definition of `TrackPlotLayoutConfig`

**Review by frasercl — APPROVED** [review]

**Pull request merged**